### DISCUSSION TITLE
Add support for fieldtype Config Fields

### DISCUSSION THREAD

**Discussion by mikemartin** [jacksleight/statamic-bard-texstyle#70] [General]
I'm attempting to make a rich "Help Text" field on my form builder using Bard and Texstyle Pins. The Bard field is appended to my fieldtype config using the code below:

```
\Statamic\Fieldtypes\Radio::appendConfigField('help_text', [
    'type' => 'bard',
    'display' => 'Help',
    'instructions' => 'Provide additional help with rich text formatting. Include links to articles, FAQs, and videos for further guidance.',
    'buttons' => [
        'bold',
        'italic',
        'anchor',
        'bts_pins'
    ],
    'bts_pins' => [
      'article',
      'faqs',
      'video'
    ]
]);
```

<img width="902" alt="Screenshot 2024-07-30 at 12 13 36 PM" src="https://github.com/user-attachments/assets/19f2278e-b289-4e34-83a3-543929340f4e">

It looks like when the bard content is rendered the Pins node looks for a parent entry to [create a Field](https://github.com/jacksleight/statamic-bard-texstyle/blob/c413785ed67316a4d5e7af200bfc2aa7a770213f/src/Nodes/Pin.php#L268). However, because this is a config field, there is no parent entry and it returns this `Call to a member function parent()` error: https://flareapp.io/share/LPd86KbP

I know this is a major edge case, but do you think there is a workaround to render Pins when used on a form config field?

**Comment by jacksleight** [comment]
> I know this is a major edge case, but do you think there is a workaround to render Pins when used on a form config field?

Hmmmm good question. Honestly never even thought to try this, but ideally it should work. I'll see if there's a fix.

**Comment by mikemartin** [comment]
Thanks for the fix. Just posting here that the issue was that config fields are not augmented, so you need to render bard with the bard_html modifier. @jacksleight pushed a fix to support the bard_html modifier.
https://github.com/jacksleight/statamic-bard-texstyle/releases/tag/3.2.3
